src/cli/commands/workspace.js

```diff
@@ -39,8 +40,13 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     throw new MessageError(reporter.lang('workspaceUnknownWorkspace', workspaceName));
   }
 
+  const workspace = workspaces[workspaceName];
+
   try {
-    await child.spawn(process.argv[0], [process.argv[1], ...rest], {stdio: 'inherit'});
+    await child.spawn(NODE_BIN_PATH, [YARN_BIN_PATH, ...rest], {
```

Comment:
I think in the future it's probably worthwhile not to use a child process at all, would save a whole bunch of overhead